src/Microsoft.Health.Fhir.Shared.Api/Controllers/SearchParameterController.cs

```diff
@@ -35,14 +37,19 @@ public class SearchParameterController : Controller
     {
         private readonly IMediator _mediator;
         private readonly CoreFeatureConfiguration _coreFeaturesConfig;
+        private bool _isAzureApiForFhir = false;
 
-        public SearchParameterController(IMediator mediator, IOptions<CoreFeatureConfiguration> coreFeatures)
+        public SearchParameterController(IMediator mediator, IOptions<CoreFeatureConfiguration> coreFeatures, IOptions<ThrottlingConfiguration> throttlingConfig)
         {
             EnsureArg.IsNotNull(mediator, nameof(mediator));
             EnsureArg.IsNotNull(coreFeatures?.Value, nameof(coreFeatures));
+            EnsureArg.IsNotNull(throttlingConfig?.Value, nameof(throttlingConfig));
 
             _mediator = mediator;
             _coreFeaturesConfig = coreFeatures.Value;
+
+            // Used to prevent Azure Api for Fhir from using this controller.
+            _isAzureApiForFhir = throttlingConfig.Value.DataStore.Equals(KnownDataStores.CosmosDb, StringComparison.OrdinalIgnoreCase);
```

Comment:
Is this the official way to check if a FHIR instance is running Gen1 or Gen2?

Reply:
It is a good question, but it seems that this may be the only place that we save the datastore value in the config objects.  I think it will work.

Reply:
No, but we do use it to determine the throttling config default for the concurrency limit between gen1 and gen2. Otherwise I'm not aware of any "official" way to determine it other than the datastore.

Reply:
I think that our solution is based on using dependency injection to solve this type of problem, and not have this type of "conditional expressions". 
PRobably, it would be the ideal to have an interface that has two different implementation per data source, and it's initialized during Startup.

Reply:
Reworked this based on Fernando's suggestion. Thanks!